functions/index.js

```diff
@@ -11,53 +11,291 @@ admin.initializeApp({
 // ==========================================================
 
 exports.signup = functions.https.onRequest((req, res) => {
+  /*
+    input = {
+      email: '1@email.com',
+      name: 'Full Name',
+      password: 'topsecret',
+      type: 'student // EITHER 'student' or 'teacher' -- ALL SMALL CASE
+    }
+  */
+  function addToDatabase(uid, type) {
+    admin.database().ref(`users/${uid}`).set({type: type})
+    .then(()=> {
```

Comment:
APIs did not pass the firebase deploy lint test.

---

/Users/Desktop/Rails/functions/index.js
24:11 error Each then() should return a value or throw promise/always-return
39:22 warning Expected to return a value at the end of arrow function consistent-return
40:9 error Each then() should return a value or throw promise/always-return
76:9 error Each then() should return a value or throw promise/always-return
111:11 error Each then() should return a value or throw promise/always-return
147:20 warning Expected to return a value at the end of arrow function consistent-return
149:9 error Each then() should return a value or throw promise/always-return
187:11 error Each then() should return a value or throw promise/always-return
229:20 warning Expected to return a value at the end of arrow function consistent-return
231:9 error Each then() should return a value or throw promise/always-return
251:11 error Each then() should return a value or throw promise/always-return
282:20 warning Expected to return a value at the end of arrow function consistent-return
284:9 error Each then() should return a value or throw promise/always-return

---

May be firebase and CI are using different lint standard? Since it pass all the test with travis.

Reply:
I think we can remove the firebase lint test.

Reply:
I dont think they should be removed the linting is there to enforce the best practices.

Reply:
Linting was never removed, code was changed to follow rules.